apps/payments/app/[locale]/(hosted)/admin/page.tsx

```diff
@@ -0,0 +1,12 @@
+import { notFound, redirect, RedirectType } from "next/navigation";
+import { routeDefinitions } from "../../../routeDefinitions";
+import { PgSessions } from "auth/sessions";
+
+export default async () => {
+  const { publicServant } = await PgSessions.get();
+
+  if (publicServant)
```

Comment:
It is better to check the negative outcome first and after all the checks let the page to do its default purpose.

```suggestion
  if (!publicServant)
    return notFound();
    
return redirect(routeDefinitions.paymentSetup.slug, RedirectType.replace);  
```

Reply:
Integrated, thanks :)